taipy/core/_entity/submittable.py

```diff
@@ -94,9 +93,9 @@ def is_ready_to_run(self) -> Reasons:
 
         for node in self.get_inputs():
             if node._edit_in_progress:
-                reason._add_reason(node.id, _build_data_node_is_being_edited_reason(node.id))
+                reason._add_reason(node.id, DataNodeEditInProgress(node.id))
```

Comment:
Since the name difference between Reasons and Reason classes is only singular or plural, I would love it if the variables to match the class name to avoid future confusion.

```suggestion
                reasons._add_reason(node.id, DataNodeEditInProgress(node.id))
```